### PR TITLE
Bugs/5548 fix geo error

### PR DESCRIPTION
**Pull request by utas-raymondng**

### CODE REVIEW

File: indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java [hunk 1]
```diff
@@ -41,49 +44,61 @@ public class StacUtils {
...
+        if(listOfPolygons != null) {
+            final Envelope overallBoundingBox = new Envelope();
+            final AtomicBoolean hasBoundingBoxUpdate = new AtomicBoolean(false);
+            listOfPolygons
+                    .forEach(polygons -> {
+                        for (Geometry polygon : polygons) {
+                            // Add polygon one by one to expand the overall bounding box area, this is requirement
+                            // of STAC to have an overall bounding box of all smaller area as the first bbox in the list.
+                            if (polygon != null) {
+                                overallBoundingBox.expandToInclude(polygon.getEnvelopeInternal());
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java:62]: should we check if `polygon.getEnvelopeInternal()` is null or empty first before flagging bounding box update to true?

File: indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java [hunk 2]
```diff
@@ -41,49 +44,61 @@ public class StacUtils {
...
-                                    BigDecimal.valueOf(c.getY()).setScale(scale, RoundingMode.HALF_UP)
-                            ));
+                for (List<Geometry> polygons : listOfPolygons) {
+
+                    if (!polygons.isEmpty()) {
+                        final Envelope individualEnvelope = new Envelope();
+
+                        for (Geometry p : polygons) {
+                            if (p != null) {
+                                individualEnvelope.expandToInclude(p.getEnvelopeInternal());
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java:84]: same here

**Review by vietnguyengit — COMMENTED** [review]
LGTM, minor comments added

File: indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java (hunk 1 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/StacUtils.java:62]: The list already check null before, but I can still add it

**Review by vietnguyengit — APPROVED** [review]
LGTM

**Pull request merged**